tests/scripts/test_config.sh

```diff
@@ -4,19 +4,15 @@
 . "$(dirname "$(realpath "$0")")/config.source"
 
 good_settings="${XDG_CONFIG_HOME}/pulp/settings.toml"
-bad_settings="${XDG_CONFIG_HOME}/pulp/bad_settings.toml"
+bad_settings="bad_settings.toml"
 test_settings="test.toml"
 
 export XDG_CONFIG_HOME=/nowhere
 
-cleanup() {
-  rm $test_settings
-}
-trap cleanup EXIT
-
-
 # SETTINGS OPTION
 
+sed -e '/base_url/c\base_url = "http://badurl"' "$good_settings" > "$bad_settings"
```

Comment:
I am pretty sure, this will match the expectations of our tests better than the static one.